src/wallet/rpcwallet.cpp

```diff
@@ -709,9 +709,18 @@ static UniValue getbalance(const JSONRPCRequest& request)
 
     LOCK2(cs_main, pwallet->cs_wallet);
 
-    const UniValue& dummy_value = request.params[0];
-    if (!dummy_value.isNull() && dummy_value.get_str() != "*") {
-        throw JSONRPCError(RPC_METHOD_DEPRECATED, "dummy first argument must be excluded or set to \"*\".");
+    const UniValue& trusted_only_uv = request.params[0];
+    bool trusted_only = true;
+    if (trusted_only_uv.isBool()) {
+        trusted_only = trusted_only_uv.get_bool();
+    } else if (trusted_only_uv.isStr()) {
+        // backward compatibility
+        if (trusted_only_uv.get_str() != "*") {
```

Comment:
I'd prefer to drop this undocumented functionality if at all possible. It's confusing that `getbalance` and `getbalance *` should return different values (and this argument has been deprecated since https://github.com/bitcoin/bitcoin/commit/7b782f5b01f4c2d906a28800d01ffd05ad257cbe).

Reply:
They always have returned different values. I think the API changes here clear up the confusion?